presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestAddExchangesPlans.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.sql.planner.assertions.BasePlanTest;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.aggregation;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.equiJoinClause;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.exchange;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.join;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.tableScan;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+import static com.facebook.presto.sql.planner.plan.ExchangeNode.Scope.REMOTE;
+import static com.facebook.presto.sql.planner.plan.ExchangeNode.Type.REPARTITION;
+import static com.facebook.presto.sql.planner.plan.JoinNode.Type.INNER;
+
+public class TestAddExchangesPlans
```

Comment:
Why not to add to `TestAddExchanges`?

Reply:
I thought about that, but TestAddExchanges is all about checking the property prioritizing logic (currently there are no plan tests for AddExchanges), so this sort of test seemed out of place. .  I don't feel strongly, though.

Reply:
I'd prefer to leave it as is or add the tests to TestAddExchanges .  This is specifically mean to be a planner test of the addExchanges rule.  And I'd like to encourage more such tests in the future :).  AddExchanges is super complicated and we don't seem to have planner tests that verify that it's doing what we expect.

Reply:
Sure.  What would make it clearer?  I want to add planner tests that test that the plan you get for the query after the optimizer runs is what you expect.  We have these kinds of tests for many other optimizers, but not AddExchanges. TestAddExchanges tests the logic of how partitioning preferences work.